lib/core/src/client/preview/client_api.js

```diff
@@ -105,13 +105,27 @@ export default class ClientApi {
 
       const fileName = m ? m.id : null;
 
-      // Add the fully decorated getStory function.
-      this._storyStore.addStory(kind, storyName, this._decorateStory(getStory, decorators), {
-        ...this._globalParameters,
-        ...localParameters,
-        ...parameters,
-        fileName,
+      const allParam = { fileName };
+
+      [this._globalParameters, localParameters, parameters].forEach(params => {
+        if (params) {
+          Object.keys(params).forEach(key => {
+            if (typeof params[key] === 'object') {
+              allParam[key] = Object.assign({ ...allParam[key] }, params[key]);
```

Comment:
Maybe just `{ ...allParam[key], ...params[key] }`? Kind of odd to use `Object.assign` and object spread ;)